create_builder.go

```diff
@@ -40,16 +42,94 @@ func (c *Client) CreateBuilder(ctx context.Context, opts CreateBuilderOptions) e
 		return errors.Wrap(err, "fetch build image")
 	}
 
+	builder, err := c.createBaseBuilder(ctx, opts, baseImage)
+	if err != nil {
+		return errors.Wrap(err, "failed to create builder")
+	}
+
+	err = c.addBuildpacksToBuilder(ctx, opts, builder)
+	if err != nil {
+		return errors.Wrap(err, "failed to add buildpacks to builder")
+	}
+
+	builder.SetOrder(opts.Config.Order)
+	builder.SetStack(opts.Config.Stack)
+
+	return builder.Save(c.logger)
+}
+
+func validateBuilderConfig(conf pubbldr.Config) error {
+	if conf.Stack.ID == "" {
+		return errors.New("stack.id is required")
+	}
+
+	if conf.Stack.BuildImage == "" {
+		return errors.New("stack.build-image is required")
+	}
+
+	if conf.Stack.RunImage == "" {
+		return errors.New("stack.run-image is required")
+	}
+
+	return nil
+}
+
+func (c Client) validateRunImageConfig(ctx context.Context, opts CreateBuilderOptions) error {
```

Comment:
we're mixing pointer and value receivers here. This should also be grouped with other `*Client` methods.

```suggestion
func (c *Client) validateRunImageConfig(ctx context.Context, opts CreateBuilderOptions) error {
```

Reply:
Why should this be a pointer receiver? This isn't changing anything on the client, it's just validating that the information from it?

Reply:
Never mind, it's much better, given what could be the size of the client struct. Will change!